activerecord/test/cases/collection_cache_key_test.rb

```diff
@@ -254,6 +252,21 @@ class CollectionCacheKeyTest < ActiveRecord::TestCase
       end
     end
 
+    test "reset will reset cache_version" do
+      with_collection_cache_versioning do
+        developers = Developer.all
+        cache_version = developers.cache_version
+
+        Developer.update_all(updated_at: Time.now.utc + 1.second)
+
+        assert_equal cache_version, developers.cache_version
+
+        developers.reset
+
+        assert_not_equal cache_version, developers.cache_version
```

Comment:
Instead of asserting that `cache_version` is simply different than it was, what about being more explicit?

```suggestion

        assert_equal Developer.all.cache_version, developers.cache_version

        Developer.update_all(updated_at: Time.now.utc + 1.second)
        developers.reset

        assert_equal Developer.all.cache_version, developers.cache_version
```

Reply:
Sounds good. Agreed, much better!